articles/cross_compilation_build_tools_design.md

```diff
@@ -0,0 +1,161 @@
+  - Those ROS 2 base images are variants of [sysroot/Dockerfile_ubuntu_arm](https://github.com/ros2/cross_compile/blob/master/sysroot/Dockerfile_ubuntu_arm) from ros2/cross_compile.
+  - OSRF would publish those base ROS 2 images, with ROS 2 pre built from source. OSRF would also publish other base ROS 2 images that install `ros-$ROS_DISTRO-desktop` with Apt, instead of installing the system dependencies for ROS 2 with `rosdep install`.
+- The plugin then sets the environment variable `SYSROOT` to the sysroot path, and other env vars required to configure the
+toolchain file [generic_linux.cmake](https://github.com/ros2/cross_compile/blob/master/cmake-toolchains/generic_linux.cmake) from ros2/cross_compile,
+and launches an instance of the `BuildVerb` class configured with `-DCMAKE_TOOLCHAIN_FILE` in `cmake-args` pointing to that toolchain file, to trigger
+a `colcon build`.
+
+The `colcon cc-build` plugin will also support the following optional arguments:
+
+- `--sysroot-base-image`: Specifies a Docker image that will be used as the base image for the workspace dependent sysroot image. This should be a string that is a valid
+argument for `docker image pull`, and that can be used in `FROM` Dockerfile statement. If it is not specified, the plugin will deduce the value of
+`--sysroot-base-image` from the `--arch` and `--os` arguments, for the architecture-OS combinations that have an image published for them on
+[osrf/ros2 on Docker Hub](https://hub.docker.com/r/osrf/ros2/).
+- `--sysroot-image`: Specifies a Docker image for the workspace sysroot. The build will export the sysroot from that image, instead of creating a sysroot image from
+the workspace. No check is performed to ensure the image has all the dependencies required by the packages in the workspace.
+- `--force-sysroot-build`: Boolean flag to force exporting the sysroot even if it already available in the workspace. This is useful for example when new system
+dependencies are added to a package in the workspace. This leads to rebuilding the dependent sysroot image after the `COPY` statements that copies the workspace into
+the image, including running `rosdep install` from scratch. This does not rebuild the base ROS 2 Docker image.
+- `--sysroot-path`: Specifies a local path for an existing sysroot. This can be useful for example if different workspaces share the same set of system dependencies.
+- `--use-base-image-sysroot`: Boolean flag, when true the base ROS 2 Docker image is used to build the sysroot, so the dependencies declared in packages in the current
+workspace are ignored. This is useful to speed up builds for packages without additional dependencies.
+
+Support for other platforms can be added by creating corresponding ROS 2 Docker images, and variants of the toolchain file if required.
+We could include support for ARM-HF and ARM64 in the first release of the plugin.
```

Comment:
Thanks for the suggestion, other comments in this PR had similar concerns. After incorporating that feedback that idea now is that the platform is determined by a tuple ` ('arch', 'os', 'distro', 'rmw')`, for example `(generic_armhf, ubuntu_bionic, ROS crystal, RMW FastRTPS)`, as seen in my previous comment. That would allow extending this in the future with support for more specific ECUs than `generic_armhf`, as suggested by @dejanpan above